solr/solrj/src/java/org/apache/solr/client/solrj/request/CollectionAdminRequest.java

```diff
@@ -1652,15 +1654,21 @@ public CreateCategoryRoutedAlias setMaxCardinality(int maxCardinality) {
       return this;
     }
 
+    public CreateCategoryRoutedAlias setMustMatch(String regex) {
+      this.mustMatch = regex;
+      return this;
+    }
+
     @Override
     public SolrParams getParams() {
       ModifiableSolrParams params = (ModifiableSolrParams) super.getParams();
       params.add(CommonParams.NAME, aliasName);
       params.add(ROUTER_TYPE_NAME, "category");
       params.add(ROUTER_FIELD, routerField);
+      params.add(ROUTER_MAX_CARDINALITY, maxCardinality.toString());
 
-      if (maxCardinality != null) {
-        params.add(ROUTER_MAX_CARDINALITY, maxCardinality.toString());
+      if (mustMatch != null) {
```

Comment:
Reading this on github, so maybe I'm reading it wrong, but won't mustMatch always start out null and this check won't ever pass?

Reply:
It only passes if setMustMatch was called,
since it sets mustMatch.